var/spack/repos/builtin/packages/libuv/package.py

```diff
@@ -2,48 +2,138 @@
+        )
+        version(
+            "1.43.0", sha256="90d72bb7ae18de2519d0cac70eb89c319351146b90cd3f91303a492707e693a4"
+        )
+        version(
+            "1.42.0", sha256="43129625155a8aed796ebe90b8d4c990a73985ec717de2b2d5d3a23cfe4deb72"
+        )
+        version(
+            "1.41.1", sha256="65db0c7f2438bc8cd48865de282bf6670027f3557d6e3cb62fb65b2e350a687d"
+        )
+        version(
+            "1.41.0", sha256="1184533907e1ddad9c0dcd30a5abb0fe25288c287ff7fee303fff7b9b2d6eb6e"
+        )
+        version(
+            "1.40.0", sha256="61a90db95bac00adec1cc5ddc767ebbcaabc70242bd1134a7a6b1fb1d498a194"
+        )
+        version(
+            "1.39.0", sha256="5c52de5bdcfb322dbe10f98feb56e45162e668ad08bc28ab4b914d4f79911697"
+        )
+        version(
+            "1.38.1", sha256="0ece7d279e480fa386b066130a562ad1a622079d43d1c30731f2f66cd3f5c647"
+        )
+        version(
+            "1.25.0", sha256="0e927ddc0f1c83899000a63e9286cac5958222f8fb5870a49b0c81804944a912"
+        )
     version("1.10.0", sha256="0307a0eec6caddd476f9cad39e18fdd6f22a08aa58103c4b0aead96d638be15e")
     version("1.9.0", sha256="d595b2725abcce851c76239aab038adc126c58714cfb572b2ebb2d21b3593842")
 
     def url_for_version(self, version):
-        if self.spec.satisfies("@:1.43"):
+        if self.spec.satisfies("@:1.43") or self.spec.satisfies("build_system=cmake"):
+            # This version includes CMake files unlike the '-dist' source distribution below
             url = "https://dist.libuv.org/dist/v{0}/libuv-v{0}.tar.gz"
         else:
             # From 1.44 on, the `-dist` download includes a configure script
             url = "https://dist.libuv.org/dist/v{0}/libuv-v{0}-dist.tar.gz"
         return url.format(version, version)
 
-    depends_on("automake", type="build", when="@:1.43.0")
-    depends_on("autoconf", type="build", when="@:1.43.0")
-    depends_on("libtool", type="build", when="@:1.43.0")
-    depends_on("m4", type="build", when="@:1.43.0")
+    # Windows needs a CMake build, but the cmake-enabled sources do not have a
+    # pre-generated configure script to enable the autotools build, so: (a)
+    # pull different sources if you are on Windows and (b) make sure cmake
+    # build is not chosen on Linux
+    # (because Linux does not download the cmake-enabled source).
+
+    # new libuv versions should only use CMake to prevent the scenario
+    # described above
+    build_system(
+        conditional("cmake", when="@1.48: platform=windows"), "autotools", default="autotools"
+    )
+
+    with when("build_system=autotools"):
+        depends_on("automake", type="build", when="@:1.43.0")
+        depends_on("autoconf", type="build", when="@:1.43.0")
+        depends_on("libtool", type="build", when="@:1.43.0")
+        depends_on("m4", type="build", when="@:1.43.0")
+
+    with when("build_system=cmake"):
+        # explictly require ownlibs
```

Comment:
I think this comment is unnecessary

Reply:
It's unfinished (oops), it was supposed to say "explicitly require ownlibs to indicate we're short circuiting the cmake<->libuv cyclic dependency here"

Reply:
If you don't find that useful I can still take it out, just thought that would be useful context

Reply:
yeah the full comment seems useful, sounds good if you expand it